### PR TITLE
[Discussion] MAN2-2876 ZXCVBN Bundling

### PR DESCRIPTION
**Pull request by rjstires**
## Purpose
It was found the ZXCVBN (390kb minified and compressed), a library used to calculate password strength, was being imported statically into the index.html. In order to resolve I created a high-order component utilizing the react-loadable library to lazily load the library as needed that could wrap a component who needed the library. I then implemented the HOC on the ChangePassword.js file as a proof of concept, updated the props to reflect the additional prop from the HOC, and finally the implementation of PasswordInput as per it's changes in linode-react-components (https://github.com/rjstires/linode-react-components/tree/MAN2-2876-zxcvbn-bundling).

There are 9 uses of the PasswordInput component, I would update each instance of it's usage with the new props and the newly created HOC.

## Notes
Dev. dependency "dynamic-import-node" module was added as required by Jest to handle the import syntax.

### CODE REVIEW

File: src/profile/components/ChangePassword.js [hunk 1]
```diff
@@ -87,4 +91,7 @@ export default class ChangePassword extends Component {
 
 ChangePassword.propTypes = {
   dispatch: PropTypes.func.isRequired,
+  calculatePasswordStrength: PropTypes.func.isRequired,
```
Comment (displague) [inline src/profile/components/ChangePassword.js]: Perhaps `passwordStrengthCalculator: PropTypes.func.isRequired`
`calculatePasswordStrength` sounds boolean

File: src/decorators/withZxcvbn.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Loadable from 'react-loadable';
+
+const withZxcvbn = (Component) => Loadable({
+  loader: () => import('zxcvbn'),
+  loading: () => null,
+  render(loaded, props) {
+    return React.createElement(Component, {
+      ...props,
+      calculatePasswordStrength: loaded,
```
Comment (asauber) [inline src/decorators/withZxcvbn.js]: should be `passwordStrengthCalculator`

**Comment by rjstires** [comment]
Hey. Don't forget to remove the build process that copies XCVBN over from node_modules.

File: src/profile/components/ChangePassword.js [hunk 3]
```diff
@@ -42,6 +43,8 @@ export default class ChangePassword extends Component {
 
   render() {
     const { password, expires, errors, loading } = this.state;
+    const { calculatePasswordStrength } = this.props;
+    const passwordStrength = !isEmpty(password) && calculatePasswordStrength(password).score;
```
Comment (asauber) [inline src/profile/components/ChangePassword.js]: We can do `(!isEmpty(password) && calculatePasswordStrength(password).score) || null` to simplify the type of `strength`

File: package.json [hunk 4]
```diff
@@ -87,7 +88,7 @@
     "history": "^4.7.2",
     "isomorphic-fetch": "^2.2.0",
     "jest": "^21.2.1",
-    "linode-components": "https://github.com/linode/linode-react-components#v0.0.7",
+    "linode-components": "https://github.com/rjstires/linode-react-components#dc65a9343c2f76445514118040469a6e8aea9a2f",
```
Comment (asauber) [inline package.json]: We should cut a release of `linode-components` and use it here when the corresponding PR is merged.

**Pull request merged**